Board format as 1-line string

jasonmm opened the pull request:
I wanted the ability to represent the board as a 1-line string.  Figured I would share it back if you find it useful.

jasonmm committed:
Moved `:use` items into `:require`.

jasonmm committed:
Added function that allows using a 1-line string as a board.

The new `boggle.utils/str->board` function converts a 1-line string
board into the multi-line format expected by `boggle.core/load-board`.

jasonmm committed:
Added test for creating a board matrix from a 1-line string.

gfarrell commented on src/boggle/core.clj:
```diff
@@ -1,8 +1,8 @@
-    (:gen-class))
+  (:require [clojure.string :as string]
+            [clojure.java.io :as io]
+            [clojure.math.numeric-tower :refer [abs]])
```
May I ask what the reasoning is behind this change?

jasonmm replied:
My motivation for the change was mostly based on [this section of the style guide](https://github.com/bbatsov/clojure-style-guide#prefer-require-over-use) (and [this post](https://stuartsierra.com/2016/clojure-how-to-ns.html#main-clauses)).  I think the motivation for preferring `:require` is because `:use` makes it easier to overwrite or shadow var names since it automatically refers them into the current namespace.

gfarrell reviewed (commented):
Hi @jasonmm, thanks for opening this PR! I just have a question about the use of `use` vs. `require`.

jasonmm committed:
Added deps.edn so the repo can be used as a dependency.

The Clojure CLI tools require dependencies referenced by Git SHA to have
a deps.edn file. See https://www.reddit.com/r/Clojure/comments/8m39p9/github_ethagnawltoolsdepsexperiments/dzl50v6/.
